src/handle_config.cpp

```diff
@@ -59,22 +59,48 @@ bool is_string_data(Ucfb_reader_strict<"DATA"_mn> data)
 
 bool is_hash_data(Ucfb_reader_strict<"DATA"_mn> data)
 {
-   const std::array<std::uint32_t, 7> hashes = {
+   bool retVal = false;
+   uint32_t a[] = {
       0x156b70a1, // GrassPatch
       0xaaea5743, // File
       0x0e0d9594, // Sound
       0xc28f0c96, // CollisionSound
       0x84874d36, // Path
       0x6850acc6, // BorderOdf
-      0x6a6fb399  // LeafPatch
+      0x6a6fb399, // LeafPatch
+      0x8d39bde6, // Name
+      0xaef39ccf, // Movie
+      0xca04efe0, // Inherit
+      0xf3c342e6, // Segment
+      0x274e1290, // Font
+      0x115bfcb9, // Subtitle
+      0x5fd56df9, // BUS
+      0x5f6f6d65,  // Stream
+      0x2e93ef4c,  /* "SoundStream" */
+      0x96e382a7,  /* "Sample" */
+      0x5fb91e8c,  /* "Group" */
+      0xab3e0bff,  /* "Class" */
+      0x2a9f6673, /* "FootstepLeftWalk" */
+      0xe79eb7c2, /* "FootstepRightWalk" */
+      0x640f6efb, /* "FootstepLeftRun" */
+      0xa4df1b00, /* "FootstepRightRun" */
+      0xb59a8702, /* "FootstepLeftStop" */
+      0x54e8af1f, /* "FootstepRightStop" */
+      0xa73f5c0d, /* "Jump" */
+      0x6dea369c, /* "Land" */
+      0x16ff3d1e, /* "Roll" */
+      0x65bfa4e1, /* "Squat" */
+      0x70dc49d0, /* "BodyFall" */
+      0xabd36816,  /* "I3DL2ReverbPreset" */
    };
-
+   
    const auto data_hash = data.read_trivial<std::uint32_t>();
    const auto element_count = data.read_trivial_unaligned<std::uint8_t>();
-
-   return ((std::find(std::cbegin(hashes), std::cend(hashes), data_hash) !=
-            std::cend(hashes)) &&
-           element_count > 0);
+   
+   if (element_count > 0) {
+      retVal = std::find(std::begin(a), std::end(a), data_hash) != std::end(a);
+   }
+   return retVal;
```

Comment:
```suggestion

   return ((std::find(std::cbegin(hashes), std::cend(hashes), data_hash) !=
            std::cend(hashes)) &&
           element_count > 0);
```
As said above I feel this is better than having a separate variable just for storing the return value in such a simple function. Also as I said above I'm happy to shift on this if there's some readability issue I'm missing.

Reply:
So, in these circumstances I will often use a variable to hold the return value to make things easier to inspect while debugging.

Reply:
Ah yeah, that makes sense. In terms of keeping it that way beyond debugging it would come down to if we expect this function to be a common stop during debugging in the future or not. If not then when needed while debugging in the future it can always be edited to assign to a variable first before returning.